src/client.ts

```diff
@@ -296,7 +296,7 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
     // set the key
     this.key = key;
     this.listeners = {};
-    this.state = new ClientState<StreamChatGenerics>();
+    this.state = new ClientState<StreamChatGenerics>({ client: this });
```

Comment:
I am wandering whether it makes sense to even have `client.state` when the client is initialized with cache disabled. But that would probably be a breaking change right?

Reply:
Not necessarily a breaking change, but I guess it's safer this way for now - I guess it's better to have it there in case the `client.state` is extended at some point with other things so that we don't have to revisit the stuff here.

Reply:
Why do we even have a attribute called `state` that keeps user cache only 🤦 .

Reply:
No idea either :D Hadn't fiddled with it before sadly, hence why it took me a while to reverse engineer everything here